Sources/SwiftDoc/Interface.swift

```diff
@@ -190,6 +190,14 @@ public final class Interface {
     public func symbols(named name: String, resolvingTypealiases: Bool) -> [Symbol] {
         symbolsGroupedByIdentifier.named(name, resolvingTypealiases: resolvingTypealiases)
     }
+
+    public func symbols(named name: String, context: String, resolvingTypealiases: Bool) -> [Symbol] {
+        let foundSymbols = symbols(named: "\(context).\(name)", resolvingTypealiases: resolvingTypealiases)
+        if !foundSymbols.isEmpty {
+            return symbols
+        }
+        return symbols(named: name, resolvingTypealiases: resolvingTypealiases)
+    }
```

Comment:
If the only place we're using this is in `linkTypes`, I think a better option would be to define an extension (`fileprivate` or otherwise) with whatever helper methods you need.

Instead of `symbols(named:context:resolvingTypealiases:)`, you might do:

```swift
extension Collection {
    var nonEmpty: Self? {
        return isEmpty ? nil : self
    }
}

let candidates = (module.interface.symbols(named: "\(context).\(name)", resolvingTypealiases: true).nonEmpty ?? module.interface.symbols(named: name, resolvingTypealiases: true).filter(symbolFilter)
```

Reply:
Agree. There's some chance that we might need something like this also at other places in the future. But then we should generalize when we have the problem and not earlier.

I like your suggestion and will apply it.